pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -215,10 +220,48 @@ func (r *ReconcileDNSZone) reconcileDNSProvider(actuator Actuator, dnsZone *hive
 	r.logger.Debug("Retrieving current state")
 	err := actuator.Refresh()
 	if err != nil {
+		if strings.Contains(err.Error(), "AccessDeniedException") {
+			var conditionMessage string
+			if submatches := accessDeniedMessageRegexp.FindStringSubmatch(err.Error()); submatches != nil {
+				// formatting error message before adding it to condition
+				// sample error message: AccessDeniedException: User: arn:aws:iam::0123456789:user/testAdmin is not authorized to perform: tag:GetResources with an explicit deny\n\tstatus code: 400, request id: d3b1e18b-52af-4836-9610-45e017476a9c
+				// message after formatting: User: arn:aws:iam::0123456789:user/testAdmin is not authorized to perform: tag:GetResources with an explicit deny
+				conditionMessage = submatches[1]
+			}
+			conds, changed := controllerutils.SetDNSZoneConditionWithChangeCheck(
+				dnsZone.Status.Conditions,
+				hivev1.InvalidCredentialsCondition,
+				corev1.ConditionTrue,
+				accessDeniedReason,
+				conditionMessage,
+				controllerutils.UpdateConditionIfReasonOrMessageChange,
+			)
+			if changed {
+				dnsZone.Status.Conditions = conds
+				if err := r.Status().Update(context.Background(), dnsZone); err != nil {
```

Comment:
Should return here when the update is successful, too. Otherwise, the function will add the "Failed to retrieve hosted zone and corresponding tags" error log entry. It would also be good to add a warning log entry when the condition is set.